examples/ros/src/smarts_ros/scripts/test_smarts_ros.py

```diff
@@ -114,11 +121,17 @@ def run_forever(self):
 
         scenario = self._init_scenario()
 
-        rospy.sleep(1)
+        rospy.sleep(5)  # hack to avoid startup race condition
```

Comment:
Yeah, the race condition is still there.  It's roughly:  the scenario path is sent to the `ros_driver` node, which receives it on a callback thread.  Then the main thread there has to cycle around (at whatever frequency has been specified by the `target_freq` parameter) to notice that the path has been set, which triggers a call to reset SMARTS.  Only _then_ will the service return the updated scenario path in the `self._smarts_info_srv()` call that follows this sleep.

Under real operating conditions, this will usually not be a problem because a user is very unlikely to query smarts for the scenario path that it _just_ _set_.  In fact, a more expected pattern might be:
`while self._smarts_info_srv().current_scenario_path != path_I_just_sent:  rospy.sleep(.2)` to basically wait for the SMARTS node to get setup before sending any messages that try to step the scenario.  However, the purpose of the `test_smarts_ros` node is, well, just that... to test the functionality in a unit-test sort of way, so that's why I think the sleep here is ok.

I agree though about the 5s being long.  I think it can be reduced, but I'm not sure how small.  I'll change it to 3s for now.

Reply:
I also made the comment a little more specific.